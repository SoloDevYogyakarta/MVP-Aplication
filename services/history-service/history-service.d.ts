/// <reference types="multer" />
import { HttpStatus } from '@nestjs/common';
import { CreateHistoryField } from '../../dto/history-dto/history-dto';
export declare class HistoryService {
    private readonly logger;
    create(id: number, desc: string, body: CreateHistoryField[], files: Express.Multer.File[]): Promise<{
        status: HttpStatus;
        message: string;
        result: import("../../database/entities/services/order-entity/order-entity").OrderInstance;
        ids: any[];
        findOne: import("../../database/entities/authenticate/user-entity/user-entity").UserInstance;
    }>;
    update(id: number, desc: string, body: CreateHistoryField[], files: Express.Multer.File[]): Promise<{
        status: HttpStatus;
        message: string;
    }>;
    destroy(id: number): Promise<{
        status: HttpStatus;
        message: string;
    }>;
}
