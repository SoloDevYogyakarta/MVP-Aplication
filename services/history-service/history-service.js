"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var HistoryService_1;
Object.defineProperty(exports, "__esModule", { value: true });
exports.HistoryService = void 0;
const common_1 = require("@nestjs/common");
const order_entity_1 = require("../../database/entities/services/order-entity/order-entity");
const user_entity_1 = require("../../database/entities/authenticate/user-entity/user-entity");
const history_entity_1 = require("../../database/entities/services/history-entity/history-entity");
const files_entity_1 = require("../../database/entities/services/files-entity/files-entity");
let HistoryService = HistoryService_1 = class HistoryService {
    constructor() {
        this.logger = new common_1.Logger(HistoryService_1.name);
    }
    async create(id, desc, body, files) {
        this.logger.log(HistoryService_1.name);
        const findOne = await user_entity_1.userEntity.findOne({ where: { id } });
        if (!findOne) {
            throw new common_1.HttpException({
                status: common_1.HttpStatus.BAD_REQUEST,
                message: 'Account not found',
            }, common_1.HttpStatus.BAD_REQUEST);
        }
        const order = await order_entity_1.orderEntity.create({ desc, user_id: findOne.id });
        order.save();
        for (const values of body) {
            const create = await history_entity_1.historyEntity.create({
                ...values,
                order_id: order.id,
            });
            create.save();
        }
        for (const file of files) {
            const filepath = `/assets${file.path.split('/assets')[1]}`;
            const f = await files_entity_1.fileEntity.create({
                originalname: file.originalname,
                type: file.mimetype.split('/')[1],
                filepath,
                order_id: order.id,
            });
            f.save();
        }
        return {
            status: common_1.HttpStatus.CREATED,
            message: 'History has been added',
            result: order,
        };
    }
    async destroy(id) {
        const findOne = await history_entity_1.historyEntity.findOne({ where: { id } });
        if (!findOne) {
            throw new common_1.HttpException({
                status: common_1.HttpStatus.BAD_REQUEST,
                message: 'History not found',
            }, common_1.HttpStatus.BAD_REQUEST);
        }
        findOne.destroy();
        return { status: common_1.HttpStatus.OK, message: 'History has been delete' };
    }
};
exports.HistoryService = HistoryService;
exports.HistoryService = HistoryService = HistoryService_1 = __decorate([
    (0, common_1.Injectable)()
], HistoryService);
//# sourceMappingURL=history-service.js.map