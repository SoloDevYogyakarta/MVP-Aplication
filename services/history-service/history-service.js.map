{"version":3,"file":"history-service.js","sourceRoot":"","sources":["../../../src/services/history-service/history-service.ts"],"names":[],"mappings":";;;;;;;;;;AAAA,2CAMwB;AACxB,mCAAoC;AACpC,6FAAyF;AACzF,8FAA0F;AAC1F,mGAA+F;AAE/F,6FAAwF;AACxF,sDAAuD;AAGhD,IAAM,cAAc,sBAApB,MAAM,cAAc;IAApB;QACY,WAAM,GAAG,IAAI,eAAM,CAAC,gBAAc,CAAC,IAAI,CAAC,CAAC;IAyI5D,CAAC;IAvIC,KAAK,CAAC,MAAM,CACV,EAAU,EACV,SAIC,EACD,IAA0B,EAC1B,KAA4B;QAE5B,IAAI,CAAC,MAAM,CAAC,GAAG,CAAC,gBAAc,CAAC,IAAI,CAAC,CAAC;QACrC,IAAI,GAAG,GAAG,EAAE,CAAC;QACb,MAAM,OAAO,GAAG,MAAM,wBAAU,CAAC,OAAO,CAAC,EAAE,KAAK,EAAE,EAAE,EAAE,EAAE,EAAE,CAAC,CAAC;QAC5D,IAAI,CAAC,OAAO,EAAE;YACZ,MAAM,IAAI,sBAAa,CACrB;gBACE,MAAM,EAAE,mBAAU,CAAC,WAAW;gBAC9B,OAAO,EAAE,mBAAmB;aAC7B,EACD,mBAAU,CAAC,WAAW,CACvB,CAAC;SACH;QACD,MAAM,KAAK,GAAG,MAAM,0BAAW,CAAC,MAAM,CAAC;YACrC,IAAI,EAAE,SAAS,CAAC,IAAI;YACpB,IAAI,EAAE,SAAS,CAAC,IAAI;YACpB,SAAS,EAAE,SAAS,CAAC,IAAI;YACzB,OAAO,EAAE,OAAO,CAAC,EAAE;SACpB,CAAC,CAAC;QACH,KAAK,CAAC,IAAI,EAAE,CAAC;QACb,KAAK,MAAM,MAAM,IAAI,IAAI,EAAE;YACzB,IAAI,IAAA,aAAI,EAAC,IAAA,aAAI,EAAC,MAAM,EAAE,CAAC,IAAI,EAAE,WAAW,EAAE,QAAQ,CAAC,CAAC,CAAC,EAAE;gBACrD,MAAM,MAAM,GAAG,MAAM,8BAAa,CAAC,MAAM,CAAC;oBACxC,GAAG,IAAA,aAAI,EAAC,MAAM,EAAE,CAAC,IAAI,EAAE,WAAW,EAAE,QAAQ,CAAC,CAAC;oBAC9C,QAAQ,EAAE,KAAK,CAAC,EAAE;iBACnB,CAAC,CAAC;gBACH,MAAM,CAAC,IAAI,EAAE,CAAC;gBACd,GAAG,CAAC,IAAI,CAAC,MAAM,CAAC,EAAE,CAAC,CAAC;aACrB;SACF;QACD,KAAK,MAAM,CAAC,KAAK,EAAE,IAAI,CAAC,IAAI,KAAK,CAAC,OAAO,EAAE,EAAE;YAC3C,MAAM,QAAQ,GAAG,UAAU,IAAI,CAAC,IAAI,CAAC,KAAK,CAAC,SAAS,CAAC,CAAC,CAAC,CAAC,EAAE,CAAC;YAC3D,MAAM,CAAC,GAAG,MAAM,yBAAU,CAAC,MAAM,CAAC;gBAChC,YAAY,EAAE,IAAI,CAAC,YAAY;gBAC/B,IAAI,EAAE,IAAI,CAAC,QAAQ,CAAC,KAAK,CAAC,GAAG,CAAC,CAAC,CAAC,CAAC;gBACjC,QAAQ;gBACR,QAAQ,EAAE,KAAK,CAAC,EAAE;gBAClB,IAAI,EAAE,IAAI,CAAC,KAAK,CAAC,EAAE,SAAS;gBAC5B,MAAM,EAAE,IAAI,CAAC,KAAK,CAAC,EAAE,MAAM;aAC5B,CAAC,CAAC;YACH,CAAC,CAAC,IAAI,EAAE,CAAC;SACV;QACD,OAAO;YACL,MAAM,EAAE,mBAAU,CAAC,OAAO;YAC1B,OAAO,EAAE,wBAAwB;YACjC,MAAM,EAAE,KAAK;YACb,GAAG;YACH,OAAO;SACR,CAAC;IACJ,CAAC;IAED,KAAK,CAAC,MAAM,CACV,EAAU,EACV,IAAY,EACZ,IAA0B,EAC1B,KAA4B;QAE5B,MAAM,OAAO,GAAG,MAAM,0BAAW,CAAC,OAAO,CAAC,EAAE,KAAK,EAAE,EAAE,EAAE,EAAE,EAAE,CAAC,CAAC;QAC7D,IAAI,CAAC,OAAO,EAAE;YACZ,MAAM,IAAI,sBAAa,CACrB,EAAE,MAAM,EAAE,mBAAU,CAAC,WAAW,EAAE,OAAO,EAAE,iBAAiB,EAAE,EAC9D,mBAAU,CAAC,WAAW,CACvB,CAAC;SACH;QACD,KAAK,MAAM,MAAM,IAAI,IAAI,EAAE;YACzB,MAAM,OAAO,GAAG,MAAM,8BAAa,CAAC,OAAO,CAAC;gBAC1C,KAAK,EAAE,EAAE,QAAQ,EAAE,OAAO,CAAC,EAAE,EAAE,EAAE,EAAE,MAAM,CAAC,EAAE,EAAE;aAC/C,CAAC,CAAC;YACH,IAAI,CAAC,OAAO,EAAE;gBACZ,MAAM,IAAI,sBAAa,CACrB;oBACE,MAAM,EAAE,mBAAU,CAAC,WAAW;oBAC9B,OAAO,EAAE,mBAAmB;iBAC7B,EACD,mBAAU,CAAC,WAAW,CACvB,CAAC;aACH;YACD,OAAO,CAAC,MAAM,CAAC,IAAA,aAAI,EAAC,MAAM,EAAE,CAAC,IAAI,EAAE,WAAW,EAAE,QAAQ,CAAC,CAAC,EAAE;gBAC1D,KAAK,EAAE,EAAE,EAAE,EAAE,MAAM,CAAC,EAAE,EAAE;aACzB,CAAC,CAAC;SACJ;QACD,IAAI,KAAK,EAAE,MAAM,EAAE;YACjB,MAAM,IAAI,GAAG,MAAM,yBAAU,CAAC,OAAO,CAAC;gBACpC,KAAK,EAAE,EAAE,QAAQ,EAAE,OAAO,CAAC,EAAE,EAAE;aAChC,CAAC,CAAC;YACH,KAAK,MAAM,MAAM,IAAI,IAAI,EAAE;gBACzB,IAAI,MAAM,CAAC,QAAQ,EAAE;oBACnB,IAAI;wBACF,IAAA,mBAAU,EAAC,QAAQ,MAAM,CAAC,QAAQ,EAAE,CAAC,CAAC;qBACvC;oBAAC,OAAO,GAAG,EAAE;qBAEb;iBACF;gBACD,MAAM,CAAC,OAAO,EAAE,CAAC;aAClB;YACD,KAAK,MAAM,CAAC,KAAK,EAAE,IAAI,CAAC,IAAI,KAAK,CAAC,OAAO,EAAE,EAAE;gBAC3C,MAAM,QAAQ,GAAG,UAAU,IAAI,CAAC,IAAI,CAAC,KAAK,CAAC,SAAS,CAAC,CAAC,CAAC,CAAC,EAAE,CAAC;gBAC3D,MAAM,CAAC,GAAG,MAAM,yBAAU,CAAC,MAAM,CAAC;oBAChC,YAAY,EAAE,IAAI,CAAC,YAAY;oBAC/B,IAAI,EAAE,IAAI,CAAC,QAAQ,CAAC,KAAK,CAAC,GAAG,CAAC,CAAC,CAAC,CAAC;oBACjC,QAAQ;oBACR,QAAQ,EAAE,OAAO,CAAC,EAAE;oBACpB,IAAI,EAAE,IAAI,CAAC,KAAK,CAAC,EAAE,SAAS;oBAC5B,MAAM,EAAE,IAAI,CAAC,KAAK,CAAC,EAAE,MAAM;iBAC5B,CAAC,CAAC;gBACH,CAAC,CAAC,IAAI,EAAE,CAAC;aACV;SACF;QACD,OAAO,CAAC,MAAM,CAAC,EAAE,IAAI,EAAE,EAAE,EAAE,KAAK,EAAE,EAAE,EAAE,EAAE,EAAE,CAAC,CAAC;QAC5C,OAAO,EAAE,MAAM,EAAE,mBAAU,CAAC,EAAE,EAAE,OAAO,EAAE,yBAAyB,EAAE,CAAC;IACvE,CAAC;IAED,KAAK,CAAC,OAAO,CAAC,EAAU;QACtB,MAAM,OAAO,GAAG,MAAM,8BAAa,CAAC,OAAO,CAAC,EAAE,KAAK,EAAE,EAAE,EAAE,EAAE,EAAE,CAAC,CAAC;QAC/D,IAAI,CAAC,OAAO,EAAE;YACZ,MAAM,IAAI,sBAAa,CACrB;gBACE,MAAM,EAAE,mBAAU,CAAC,WAAW;gBAC9B,OAAO,EAAE,mBAAmB;aAC7B,EACD,mBAAU,CAAC,WAAW,CACvB,CAAC;SACH;QACD,OAAO,CAAC,OAAO,EAAE,CAAC;QAClB,OAAO,EAAE,MAAM,EAAE,mBAAU,CAAC,EAAE,EAAE,OAAO,EAAE,yBAAyB,EAAE,CAAC;IACvE,CAAC;CACF,CAAA;AA1IY,wCAAc;yBAAd,cAAc;IAD1B,IAAA,mBAAU,GAAE;GACA,cAAc,CA0I1B"}