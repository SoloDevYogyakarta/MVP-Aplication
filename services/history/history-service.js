"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.HistoryService = void 0;
const common_1 = require("@nestjs/common");
const user_entity_1 = require("../../database/entities/authenticates/user-entity/user-entity");
const history_entity_1 = require("../../database/entities/services/history-entity/history-entity");
let HistoryService = class HistoryService {
    async destroy(public_id, user_id) {
        await this.isAdmin(user_id);
        const findOne = await history_entity_1.historyEntity.findOne({ where: { public_id } });
        if (!findOne) {
            throw new common_1.HttpException({
                status: common_1.HttpStatus.NOT_FOUND,
                message: 'Not Found',
            }, common_1.HttpStatus.NOT_FOUND);
        }
        findOne.destroy();
        return { status: common_1.HttpStatus.OK, message: 'History has been delete' };
    }
    async isAdmin(public_id) {
        const findOne = await user_entity_1.userEntity.findOne({ where: { public_id } });
        if (findOne?.role !== 'admin') {
            throw new common_1.HttpException('false', common_1.HttpStatus.INTERNAL_SERVER_ERROR);
        }
    }
};
exports.HistoryService = HistoryService;
exports.HistoryService = HistoryService = __decorate([
    (0, common_1.Injectable)()
], HistoryService);
//# sourceMappingURL=history-service.js.map