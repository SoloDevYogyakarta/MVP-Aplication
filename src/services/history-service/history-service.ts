import { HttpException, HttpStatus, Injectable, Logger } from '@nestjs/common';
import { userEntity } from '../../database/entities/authenticate/user-entity/user-entity';
import { historyEntity } from '../../database/entities/services/history-entity/history-entity';
import { CreateHistoryField } from '../../dto/history-dto/history-dto';

@Injectable()
export class HistoryService {
  private readonly logger = new Logger(HistoryService.name);

  async create(public_id: string, body: CreateHistoryField[]) {
    this.logger.log(HistoryService.name);
    const findOne = await userEntity.findOne({ where: { public_id } });
    if (!findOne) {
      throw new HttpException(
        {
          status: HttpStatus.BAD_REQUEST,
          message: 'Account not found',
        },
        HttpStatus.BAD_REQUEST,
      );
    }
    for (const values of body) {
      const create = await historyEntity.create({
        ...values,
        user_id: findOne.public_id,
      });
      create.save();
    }
    return { status: HttpStatus.CREATED, message: 'History has been added' };
  }

  async destroy(public_id: string) {
    const findOne = await historyEntity.findOne({ where: { public_id } });
    if (!findOne) {
      throw new HttpException(
        {
          status: HttpStatus.BAD_REQUEST,
          message: 'History not found',
        },
        HttpStatus.BAD_REQUEST,
      );
    }
    findOne.destroy();
    return { status: HttpStatus.OK, message: 'History has been delete' };
  }
}
