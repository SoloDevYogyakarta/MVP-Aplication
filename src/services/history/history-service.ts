import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
import { userEntity } from '../../database/entities/authenticates/user-entity/user-entity';
import { historyEntity } from '../../database/entities/services/history-entity/history-entity';

@Injectable()
export class HistoryService {
  async destroy(public_id: string, user_id: string) {
    await this.isAdmin(user_id);
    const findOne = await historyEntity.findOne({ where: { public_id } });
    if (!findOne) {
      throw new HttpException(
        {
          status: HttpStatus.NOT_FOUND,
          message: 'Not Found',
        },
        HttpStatus.NOT_FOUND,
      );
    }
    findOne.destroy();
    return { status: HttpStatus.OK, message: 'History has been delete' };
  }

  async isAdmin(public_id: string): Promise<void> {
    const findOne = await userEntity.findOne({ where: { public_id } });
    if (findOne?.role !== 'admin') {
      throw new HttpException('false', HttpStatus.INTERNAL_SERVER_ERROR);
    }
  }
}
