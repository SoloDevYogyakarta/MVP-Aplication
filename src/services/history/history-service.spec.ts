import { HttpStatus } from '@nestjs/common';
import { HistoryService } from './history-service';

describe('HistoryService', () => {
  let service: HistoryService;

  beforeEach(() => {
    service = new HistoryService();
  });

  it('should to be defined', () => expect(service).toBeDefined());

  it('render correctly', () => expect(service).toMatchSnapshot());

  it('destroy', async () => {
    try {
      await service.destroy('dqwdq', 'dqwdq');
    } catch (err) {
      expect({ status: err.status, message: err.message }).toEqual({
        status: HttpStatus.INTERNAL_SERVER_ERROR,
        message: 'false',
      });
    }
  });

  it('invalid admin', async () => {
    try {
      await service.isAdmin('dqdq');
    } catch (err) {
      expect({ status: err.status, message: err.message }).toEqual({
        status: HttpStatus.INTERNAL_SERVER_ERROR,
        message: 'false',
      });
    }
  });
});
