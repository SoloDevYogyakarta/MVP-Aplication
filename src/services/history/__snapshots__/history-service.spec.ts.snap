// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryService render correctly 1`] = `HistoryService {}`;
